tools/changelog-generator/changelog.js

```diff
@@ -0,0 +1,160 @@
+        const created_at = new Date(pr.created_at);
+
+        // Load the files for this PR using the API
+        const pr_files = await octokit.paginate(
+          octokit.rest.pulls.listFiles,
+          {
+            owner,
+            repo,
+            pull_number: pr.number,
+          },
+          (response) => response.data,
+        );
+
+        // Convert app, _src, /hub/ etc paths to URLs
+        affected_files = (
+          await convertFilePathsToUrls(pr_files, { skip_nav_files: true })
+        ).map((v) => {
+          return {
+            url: `https://docs.konghq.com${v.url}`,
+            status: v.status,
+          };
+        });
+
+        return {
+          number: pr.number,
+          title: pr.title,
+          description: extractDescription(pr.body),
+          user: pr.user.login,
+          url: pr.html_url,
+          labels: pr.labels,
+          created_at,
+          week: format(created_at, "I"),
+          affected_files,
+        };
+      }),
+    )
+  )
+    .filter((pr) => pr.affected_files.length > 0)
+    .filter((pr) => !pr.labels.includes("skip-changelog"));
+
+  prsWithFiles = groupBy(prsWithFiles, "week");
+
+  // Write the changelog
+  let changelogContent = `# Changelog\n\n`;
+
+  for (let week of Object.keys(prsWithFiles).reverse()) {
+    changelogContent += `## Week ${week}\n\n`;
+
+    for (const pr of prsWithFiles[week]) {
+      changelogContent += `### [${pr.title}](${pr.url}) (${format(
+        pr.created_at,
+        "yyyy-MM-dd",
+      )})\n\n`;
+
+      changelogContent += `${pr.description}\n\n`;
+
+      const addedFiles = pr.affected_files.filter((u) => u.status == "added");
+      const changedFiles = pr.affected_files.filter(
+        (u) => u.status == "modified",
+      );
+
+      if (addedFiles.length) {
+        changelogContent += `## Added\n\n`;
+        for (const file of addedFiles) {
+          changelogContent += `- ${file.url}\n`;
+        }
+      }
+
+      if (addedFiles.length && changedFiles.length) {
+        changelogContent += `\n`;
+      }
+
+      if (changedFiles.length) {
+        changelogContent += `## Modified\n\n`;
+        for (const file of changedFiles) {
+          changelogContent += `- ${file.url}\n`;
+        }
+      }
+
+      changelogContent += `\n\n`;
+    }
+  }
+
+  changelogContent = changelogContent.trim();
+
+  // Prepend changelogContent to changelog.md
+  const changelogFile = __dirname + "/../../changelog.md";
+  let existingChangelog = "";
+
+  try {
+    existingChangelog = await fs.readFile("changelog.md", "utf8");
+  } catch (e) {
+    // No file to read
+  }
+
+  changelogContent = `${changelogContent}\n\n${existingChangelog.replace(
+    "# Changelog\n\n",
+    "",
+  )}`;
+  await fs.writeFile(changelogFile, changelogContent);
+})();
+
+function extractDescription(body) {
+  body = body || "";
+  return body
+    .split("### Testing instructions")[0]
+    .replace("### Description", "")
+    .trim();
```

Comment:
> Will this still work if the PR has no ### Description section (ie someone ctrl+a and deleted the whole thing to write their own)?
> For example, some people leave the instructional text in, in



So that part of the code grabs the area between the description and the testing instructions and uses that to create the description. So if you have a PR template like: 

```
### Description

What did you change and why?
 This is what I changed!
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

JIRA: https:link.com

### Testing instructions 

test 123
```
You would end up with a PR body like

```
What did you change and why?
 This is what I changed!
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

JIRA: https:link.com
```

That then gets written into the template as 

```
# Changelog

## Week X

### [PR Title](PR URL) (PR Creation Date)

What did you change and why?
 
This is what I changed!
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

JIRA: https:link.com

```

Reply:
what about the second half of my comment, re: adjusting the PR template to standardize these entries? I get what it grabs, I'm saying that we need better descriptions to make that changelog actually readable.